pycode/memilio-epidata/memilio/epidata/transformMobilityData.py

```diff
@@ -0,0 +1,149 @@
+    under the given directory into a pandas data frame.
+
+    @param directory Path to folder where data is read.
+    @param mobility_file Mobility matrix file which has to be updated.
+    @return Mobility matrix data frame.
+    """
+    mobility_matrix = pd.read_csv(
+        directory + mobility_file + '.txt', sep=' ', header=None)
+
+    return mobility_matrix
+
+
+def createFederalStatesMobility(directory, mobility_file, file_format=dd.defaultDict['file_format']):
+    """! Creates mobility matrices for German federal states based on
+    county mobility. If mobility matrix dimensions are different from the
+    number of German counties, nothing is done.
+
+    @param directory Path to folder where data is read and written.
+    @param mobility_file Mobility matrix file which has to be updated.
+    @param file_format File format which is used for writing the data. 
+        Default defined in defaultDict.
+    @return State-aggregated mobility matrix if input matrix is sized correctly, zero otherwise.
+    """
+    mobility_matrix = getMobilityFromFile(directory, mobility_file)
+
+    if(len(mobility_matrix) == len(geoger.get_county_ids())):
+        # get county and state IDs
+        countyIDs = geoger.get_county_ids()
+        stateIDs = geoger.get_state_ids()
+        # get state ID to county ID map
+        stateID_to_countyID = geoger.get_stateid_to_countyids_map()
+        # initialize federal states mobility matrix
+        mobility_matrix_states = np.zeros(
+            (len(stateID_to_countyID), len(stateID_to_countyID)))
+
+        # iterate over state_to_county map and replace IDs by numbering 0, ..., n
+        state_indices = []
+        county_indices = []
+        for state, counties in stateID_to_countyID.items():
+            state_indices.append(stateIDs.index(state))
+            county_indices.append(
+                np.array([countyIDs.index(county) for county in counties]))
+
+        state_to_county = dict(zip(state_indices, county_indices))
+        # iterate over all states
+        for state, counties in state_to_county.items():
+            # iterate over all neighbors
+            for state_neighb, counties_neighb in state_to_county.items():
+
+                if state != state_neighb:
+
+                    mobility_matrix_states[state, state_neighb] = mobility_matrix.iloc[counties, counties_neighb].sum(
+                        axis=0).sum()
+                    mobility_matrix_states[state_neighb, state] = mobility_matrix.iloc[counties_neighb, counties].sum(
+                        axis=1).sum()
+
+        mobility_matrix_states = pd.DataFrame(mobility_matrix_states)
+        mobility_matrix_states.to_csv(
+            directory + mobility_file + '_states.txt', sep=' ', header=None, index=False)
+        return mobility_matrix_states
+
+    else:
+        return pd.DataFrame()
+
+
+def updateMobility2022(directory, mobility_file, file_format=dd.defaultDict['file_format']):
```

Comment:
File_format removed.I'd like to not provide defaults for the other two since directory default is still to be discussed and also the name of the mobility file should be discussed in the near future.